src/ui.cpp

```diff
@@ -308,7 +309,9 @@ input_context uilist::create_main_input_context() const
     }
     ctxt.register_action( "SELECT" );
     ctxt.register_action( "MOUSE_MOVE" );
-    ctxt.register_action( "CONFIRM" );
+    if( allow_confirm ) {
+        ctxt.register_action( "CONFIRM" );
+    }
```

Comment:
```suggestion
    if( allow_confirm ) {
        ctxt.register_action( "CONFIRM" );
        ctxt.register_action( "SELECT" );
    }
```
This is up to you, clicking can be treated the same as "CONFIRM" here